modules/fee/handler.go

```diff
@@ -11,76 +11,67 @@ import (
 // NameFee - namespace for the fee module
 const NameFee = "fee"
 
-// AccountChecker - interface used by SimpleFeeHandler
-type AccountChecker interface {
-	// Get amount checks the current amount
-	GetAmount(store state.KVStore, addr basecoin.Actor) (coin.Coins, error)
-
-	// ChangeAmount modifies the balance by the given amount and returns the new balance
-	// always returns an error if leading to negative balance
-	ChangeAmount(store state.KVStore, addr basecoin.Actor, coins coin.Coins) (coin.Coins, error)
+// Bank is a default location for the fees, but pass anything into
+// the middleware constructor
+var Bank = basecoin.Actor{App: NameFee, Address: []byte("bank")}
+
+// SimpleFeeMiddleware - middleware for fee checking, constant amount
+// It used modules.coin to move the money
+type SimpleFeeMiddleware struct {
+	MinFee    coin.Coin //
+	Collector basecoin.Actor
```

Comment:
Awesome that the collector is an actor - can define unique mechanisms for redistribution of fees with this. - Still need to understand how we make the fees get slashed using the basecoin.Actor